File: c++/tempodejogocomminutos.cpp
#include <stdio.h>

int main() {
    int Hora_inicio, final, minuto_inicio, minuto_final, tempo_horatotal, tempo_minutototal;
    scanf("%d %d %d %d", &Hora_inicio, &minuto_inicio, &final, &minuto_final);

    tempo_horatotal = final - Hora_inicio;

    if (tempo_horatotal < 0) tempo_horatotal = 24 + (final - Hora_inicio);

    tempo_minutototal = minuto_final - minuto_inicio;
    if (tempo_minutototal < 0){
        tempo_minutototal = 60 + (minuto_final - minuto_inicio);
        tempo_horatotal--;
    }
    if (Hora_inicio == final && minuto_inicio == minuto_final) printf("O JOGO DUROU 24 HORA(S) E 0 MINUTO(S)\n");
    else printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", tempo_horatotal, tempo_minutototal);
    return 0;
}

File: python/tempo_de_jogo_com_minutos.py
from datetime import datetime

a = input().strip().split(" ")

a1 = "{}:{}".format(a[0],a[1])
a2 = "{}:{}".format(a[2],a[3])

t1 = datetime.strptime(str(a1), "%H:%M")
t2 = datetime.strptime(str(a2), "%H:%M")

t = (t2-t1).total_seconds()

h = int(t/3600)
m = int((t-(h*3600))/60)

if h == 0 and m == 0:
    print("O JOGO DUROU 24 HORA(S) E 0 MINUTO(S)")
else:
    print("O JOGO DUROU {} HORA(S) E {} MINUTO(S)".format(h, m))